tests/integration/targets/azure_rm_hdinsightcluster/tasks/main.yml

```diff
@@ -0,0 +1,223 @@
+- name: Prepare random number
+  set_fact:
+    rpfx: "{{ resource_group | hash('md5') | truncate(7, True, '') }}{{ 1000 | random }}"
+  run_once: yes
+
+- name: Create storage account
+  azure_rm_storageaccount:
+      resource_group: "{{ resource_group }}"
+      name: "storage{{ rpfx }}"
+      account_type: Standard_LRS
+      location: eastus2
+
+- name: Sample for Azure REST API - StorageAccounts_ListKeys
+  azure_rm_resource:
+    api_version: '2018-07-01'
+    method: POST
+    resource_group: "{{ resource_group }}"
+    provider: storage
+    resource_type: storageaccounts
+    resource_name: "storage{{ rpfx }}"
+    subresource:
+      - type: listkeys
+  register: storage_output
+
+- debug:
+    var: storage_output
+
+- name: Create instance of Cluster -- check mode
+  azure_rm_hdinsightcluster:
+    resource_group: "{{ resource_group }}"
+    name: "cluster{{ rpfx }}"
+    location: eastus2
+    cluster_version: 3.6
+    os_type: linux
+    tier: standard
+    cluster_definition:
+      kind: spark
+      gateway_rest_username: http-user
+      gateway_rest_password: MuABCPassword!!@123 
+    storage_accounts:
+      - name: storage{{ rpfx }}.blob.core.windows.net 
+        is_default: yes
+        container: "cluster{{ rpfx }}"
+        key: "{{ storage_output['response']['keys'][0]['value'] }}"
+    compute_profile_roles:
+      - name: headnode
+        target_instance_count: 1
+        vm_size: Standard_D3
```

Comment:
smaller size?

Reply:
that's the smallest possible size already.